src/test/java/org/opensearch/security/support/Base64HelperTest.java

```diff
@@ -41,67 +42,121 @@ private static final class NotSafeSerializable implements Serializable {
         private static final long serialVersionUID = 5135559266828470092L;
     }
 
-    private static Serializable ds(Serializable s) {
-        return deserializeObject(serializeObject(s));
+    private static Serializable ds(Serializable s, PackageBehavior packageBehavior) {
+        return deserializeObject(serializeObject(s, packageBehavior));
     }
 
     @Test
     public void testString() {
         String string = "string";
-        Assert.assertEquals(string, ds(string));
+        Assert.assertEquals(string, ds(string, PackageBehavior.NONE));
     }
 
     @Test
     public void testInteger() {
         Integer integer = Integer.valueOf(0);
-        Assert.assertEquals(integer, ds(integer));
+        Assert.assertEquals(integer, ds(integer, PackageBehavior.NONE));
     }
 
     @Test
     public void testDouble() {
         Double number = Double.valueOf(0.);
-        Assert.assertEquals(number, ds(number));
+        Assert.assertEquals(number, ds(number, PackageBehavior.NONE));
     }
 
     @Test
     public void testInetSocketAddress() {
         InetSocketAddress inetSocketAddress = new InetSocketAddress(0);
-        Assert.assertEquals(inetSocketAddress, ds(inetSocketAddress));
+        Assert.assertEquals(inetSocketAddress, ds(inetSocketAddress, PackageBehavior.NONE));
     }
 
     @Test
     public void testPattern() {
         Pattern pattern = Pattern.compile(".*");
-        Assert.assertEquals(pattern.pattern(), ((Pattern) ds(pattern)).pattern());
+        Assert.assertEquals(pattern.pattern(), ((Pattern) ds(pattern, PackageBehavior.NONE)).pattern());
     }
 
     @Test
     public void testUser() {
         User user = new User("user");
-        Assert.assertEquals(user, ds(user));
+        Assert.assertEquals(user, ds(user, PackageBehavior.NONE));
     }
 
     @Test
     public void testSourceFieldsContext() {
         SourceFieldsContext sourceFieldsContext = new SourceFieldsContext(new SearchRequest(""));
-        Assert.assertEquals(sourceFieldsContext.toString(), ds(sourceFieldsContext).toString());
+        Assert.assertEquals(sourceFieldsContext.toString(), ds(sourceFieldsContext, PackageBehavior.NONE).toString());
     }
 
     @Test
     public void testHashMap() {
         HashMap map = new HashMap();
-        Assert.assertEquals(map, ds(map));
+        Assert.assertEquals(map, ds(map, PackageBehavior.NONE));
     }
 
     @Test
     public void testArrayList() {
         ArrayList list = new ArrayList();
-        Assert.assertEquals(list, ds(list));
+        Assert.assertEquals(list, ds(list, PackageBehavior.NONE));
+    }
+
+    @Test
+    public void testStringWithRewriteOdfePackage() {
+        String string = "string";
+        Assert.assertEquals(string, ds(string, PackageBehavior.REWRITE_AS_ODFE));
+    }
```

Comment:
Visually, these tests don't seem much different, except the PackageBehaviour parameter. Is this to test that serialization/deserialization is not affected by packageBehaviour?

Reply:
That's what I was thinking. This makes sure that serialize and deserialize works both with and without package re-write.